engine/runtime/src/bench/java/org/enso/interpreter/bench/benchmarks/semantic/WarningBenchmarks.java

```diff
@@ -0,0 +1,111 @@
+package org.enso.interpreter.bench.benchmarks.semantic;
+
+import org.enso.interpreter.test.TestBase;
+import org.enso.polyglot.MethodNames;
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.Source;
+import org.graalvm.polyglot.Value;
+import org.openjdk.jmh.annotations.*;
+import org.openjdk.jmh.infra.BenchmarkParams;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.nio.file.Paths;
+import java.util.Objects;
+import java.util.concurrent.TimeUnit;
+
+@BenchmarkMode(Mode.AverageTime)
+@Fork(1)
+@Warmup(iterations = 5, time = 1)
+@Measurement(iterations = 3, time = 3)
+@OutputTimeUnit(TimeUnit.MILLISECONDS)
+@State(Scope.Benchmark)
+public class WarningBenchmarks extends TestBase {
+    private static final int INPUT_VEC_SIZE = 10000;
+    private Context ctx;
+    private Value vecSumBench;
+
+    private Value createVec;
+    private Value noWarningsVec;
+    private Value sameWarningVec;
+
+    private Value elem;
+
+    private Value elemWithWarning;
+
+    private OutputStream out = new ByteArrayOutputStream();
+    private String benchmarkName;
+
+    @Setup
+    public void initializeBench(BenchmarkParams params) throws IOException {
+        ctx = Context.newBuilder("enso")
+                .allowAllAccess(true)
+                .logHandler(out)
+                .out(out)
+                .err(out)
+                .allowIO(true)
+                .allowExperimentalOptions(true)
+                .option(
+                        "enso.languageHomeOverride",
+                        Paths.get("../../distribution/component").toFile().getAbsolutePath()
+                )
+                .option("engine.MultiTier", "true")
+                .option("engine.BackgroundCompilation", "true")
```

Comment:
For an unknown reason, we set `engine.BackgroundCompilation` to `false` as the default, even in benchmarks. Here https://github.com/enso-org/enso/blob/develop/build.sbt#L1029. Good that you override it here, but we should maybe get rid of the default option in build.sbt?

Reply:
I think @JaroslavTulach made this possible recently in https://github.com/enso-org/enso/pull/6335 so maybe that was just forgotten there?